x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/AggregationResultUtils.java

```diff
@@ -409,6 +416,22 @@ public Object value(Object key, String type) {
 
     static class DefaultBucketKeyExtractor implements BucketKeyExtractor {
 
+        @Override
+        public Object value(Object key, String type) {
+            if (isNumericType(type) && key instanceof Double) {
+                return dropFloatingPointComponentIfTypeRequiresIt(type, (Double) key);
+            } else if ((DateFieldMapper.CONTENT_TYPE.equals(type) || DateFieldMapper.DATE_NANOS_CONTENT_TYPE.equals(type))
+                && key instanceof Long) {
+                    return DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER.formatMillis((Long) key);
```

Comment:
You use `formatMillis` here for formatting but in the line above you check that the `type` is `DateFieldMapper.DATE_NANOS_CONTENT_TYPE`.
Please double-check it works as expected in case of nanos.

Reply:
:+1: checked, I will create a code comment